app/views/previews/show.html.erb

```diff
@@ -10,22 +10,21 @@
         <%= @dataset.title %>
         <span class="heading-secondary dgu-home-spacer__bottom"><%= @datafile.name %></span>
       </h1>
+
       <% if @datafile.csv? && @datafile.preview.exists? %>
         <p>You're previewing the first <%= @datafile.preview.line_count %> rows of this file.</p>
-      <% end %>
-      <%= link_to t('.download_file'),
-      @datafile.url,
-        class: 'button',
-        :data => {
-          :'ga-event' => 'preview',
-          :'ga-format' => (@datafile.format.presence || 'n/a').upcase,
-          :'ga-publisher' => @dataset.organisation.name
-        }
+        <%= link_to t('.download_file'),
```

Comment:
We're nesting the download button in a `@datafile.csv?` condition which made me think, "if I previewed a non-csv file I wouldn't be able to download it" 🤔 

The view has upstream knowledge of how it is used, and is therefore coupled to the following condition:

https://github.com/alphagov/datagovuk_find/blob/562238f5b3d2dc72101ff2523f2df59e839c8f41/app/views/datasets/_datafile_table.html.erb#L40-L55

`Datafile` will attempt to generate a preview for any type of file:

https://github.com/alphagov/datagovuk_find/blob/562238f5b3d2dc72101ff2523f2df59e839c8f41/app/models/datafile.rb#L53-L55

But `Preview` only handles CSV files:

https://github.com/alphagov/datagovuk_find/blob/562238f5b3d2dc72101ff2523f2df59e839c8f41/app/models/preview.rb#L32

By using the `<% if @datafile.csv? && @datafile.preview.exists? %>` condition we're effectively leaking knowledge of the private behaviour of `Preview`.

This doesn't directly effect us now, but it makes it harder to add additional previews in the future.

Reply:
I think ALL of the preview stuff needs looking at when we need to support more than just CSV, but also we need to fix the `Preview` model because it is downloading the entire CSV into RAM if the remote server doesn't support HTTP-Range.  Although most CSV are less than 10k, some are very large.  I'll add a ticket for this.

Reply:
👍

Reply:
Thanks Andrew. Previews all need a refactor I agree. One option I discussed with laurent was if we run a link checker task periodically on the datafile URL's, and in Publish when the url is added, we can store  in the db `broken: true` or `false` then we wouldn't have to check the format or call `.preview` on each instance of `Datafile` either. Anyhow I'll make a ticket for it as it's out of scope of this PR.

Reply:
👍

Reply:
[trello ticket #384](https://trello.com/c/YIbDaNb8/384-linkchecker-for-datafile-urls)
[trello ticket #368](https://trello.com/c/pGP6cTkm/368-find-preview-assumes-everyone-support-http-range)